File: algorithms/2089.find-target-indices-after-sorting-array/solution.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def target_indices(nums, target)
  target_count, smaller_count = 0, 0

  nums.each do |n|
    target_count += 1 if target == n
    smaller_count += 1 if target > n
  end

  (smaller_count...smaller_count + target_count).to_a
end

File: algorithms/2089.find-target-indices-after-sorting-array/solution.go
func targetIndices(nums []int, target int) []int {
	var targetCount int
	var smallerCount int

	for _, n := range nums {
		if target == n {
			targetCount++
		} else if target > n {
			smallerCount++
		}
	}

	var res []int
	for i := smallerCount; i < smallerCount+targetCount; i++ {
		res = append(res, i)
	}

	return res
}
